### DISCUSSION TITLE
Suggestion: use basic_staj_event::length_ for arrays/objects of known size

### DISCUSSION THREAD

**Discussion by ecorm** [danielaparker/jsoncons#375] [Ideas] [answered]
I'm using the cursor interface to decode JSON, MsgPack, CBOR, etc into my own variant type (similar to your `basic_json`). For the `staj_event_type::begin_array` and `staj_event_type::begin_object` events, it would be useful if `basic_staj_event::length_` contained the number of elements when this is known (e.g. MsgPack and CBOR definite arrays/maps). This would allow me to `reserve` the data structures that will contain the elements, thus avoiding reallocations. A `length_` of zero could be a special value that means "unknown length".

**Comment by danielaparker** [comment, marked as answer]
That's a good suggestion, it was easy to implement, and is currently on master.

I've added a `size()` accessor function to `basic_staj_event`. If the event type is a `key` or a `string_value` or a `byte_string_value`, 
it returns the size of the key or string or byte string value. If the event type is a `begin_object` or a `begin_array`, it returns the size of the object or array if known, otherwise 0. For all other event types, it returns 0.
